quil-rs/src/expression/mod.rs

```diff
@@ -437,49 +437,36 @@ impl FromStr for Expression {
     }
 }
 
-static FORMAT_REAL_OPTIONS: Lazy<WriteFloatOptions> = Lazy::new(|| {
-    WriteFloatOptions::builder()
-        .negative_exponent_break(NonZeroI32::new(-5))
-        .positive_exponent_break(NonZeroI32::new(15))
-        .trim_floats(true)
-        .build()
-        .expect("options are valid")
-});
-
-static FORMAT_IMAGINARY_OPTIONS: Lazy<WriteFloatOptions> = Lazy::new(|| {
-    WriteFloatOptions::builder()
-        .negative_exponent_break(NonZeroI32::new(-5))
-        .positive_exponent_break(NonZeroI32::new(15))
-        .trim_floats(false) // Per the quil spec, the imaginary part of a complex number is always a floating point number
-        .build()
-        .expect("options are valid")
-});
-
 /// Format a num_complex::Complex64 value in a way that omits the real or imaginary part when
 /// reasonable. That is:
 ///
 /// - When imaginary is set but real is 0, show only imaginary
 /// - When imaginary is 0, show real only
 /// - When both are non-zero, show with the correct operator in between
+/// 
+/// `ryu` formats exponents from -5 to 15 in standard notation, and larger magnitudes in scientific
+/// notation.
 #[inline(always)]
 fn format_complex(value: &Complex64) -> String {
-    const FORMAT: u128 = format::STANDARD;
     if value.re == 0f64 && value.im == 0f64 {
         "0".to_owned()
-    } else if value.im == 0f64 {
-        to_string_with_options::<_, FORMAT>(value.re, &FORMAT_REAL_OPTIONS)
     } else if value.re == 0f64 {
-        to_string_with_options::<_, FORMAT>(value.im, &FORMAT_IMAGINARY_OPTIONS) + "i"
+        ryu::Buffer::new().format(value.im).to_owned() + "i"
     } else {
-        let mut out = to_string_with_options::<_, FORMAT>(value.re, &FORMAT_REAL_OPTIONS);
-        if value.im > 0f64 {
-            out.push('+')
+        let mut out = ryu::Buffer::new().format(value.re).to_owned();
+        match value.im {
```

Comment:
I don't mind going back to the `else if`/`else` here, but this seems slightly nicer to me, personally.